guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/RangeSerializer.java

```diff
@@ -82,7 +100,7 @@ public JsonSerializer<?> createContextual(SerializerProvider prov,
             endpointSer = ((ContextualSerializer) endpointSer).createContextual(prov, property);
         }
         if ((endpointSer != _endpointSerializer) || (nameMapping != null)) {
-            return new RangeSerializer(_rangeType, endpointSer, nameMapping);
+            return new RangeSerializer(_rangeType, endpointSer, nameMapping, shape);
```

Comment:
similarly, see if `shape` has changed and only create if one of settings has changed. Minor optimization, not a big deal.